metadata-ingestion/src/datahub/cli/ingest_cli.py

```diff
@@ -144,20 +141,44 @@ def run_pipeline_to_completion(
                 ret = pipeline.pretty_print_summary(warnings_as_failure=strict_warnings)
                 return ret
 
-    async def run_pipeline_async(pipeline: Pipeline) -> int:
-        loop = asyncio._get_running_loop()
-        return await loop.run_in_executor(
-            None, functools.partial(run_pipeline_to_completion, pipeline)
+    # main function begins
```

Comment:
It's a bit awkward that half the function is here and half of it is below `run_ingestion_and_check_upgrade`. Can we put the pieces together?

Reply:
Not really since we want ingestion to run parallel with upgrade checking, but neither to run if pipeline creation fails

Reply:
Does the location of the definition of the async method actually do anything? They're just nested inside `run` because we need a couple values in scope right?

Reply:
Yup